### PR TITLE
Modules

### PR DESCRIPTION
**Pull request by ank42**

### CODE REVIEW

File: l2/main.tf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module "ec2" {
+  source = "../modules/ec2"
+
+
+}
+
```
Comment (vladimirmukhin) [inline l2/main.tf]:
```suggestion
module "ec2" {
  source = "../modules/ec2"
}
```

File: modules/vpc/main.tf [hunk 2]
```diff
@@ -0,0 +1,99 @@
+
+resource "aws_vpc" "main" {
+  cidr_block = var.vpc_cidr
+}
+
+variable "public_subnet_ids" {
+  default = ["10.0.0.0/24", "10.0.1.0/24"]
```
Comment (vladimirmukhin) [inline modules/vpc/main.tf]: You are relying on defaults too much. Instead, I'd rather remove defaults and pass required variables via l1/main.tf
Same for any other variable that you pass inside the module

File: modules/vpc/output.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+output "public_subnet_id" {
+  description = "Public Subnet ID"
+  value = aws_subnet.Public.*.id
```
Comment (vladimirmukhin) [inline modules/vpc/output.tf]: You need to run `terraform fmt --recursive` in order to align style

File: modules/vpc/variable.tf [hunk 4]
```diff
@@ -0,0 +1,5 @@
+variable "vpc_cidr" {
+  type        = string
+  description = "The IP range to use for the VPC"
+  default     = "10.0.0.0/16"
+}
```
Comment (vladimirmukhin) [inline modules/vpc/variable.tf]: Trailing newline is missing

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: l2/main.tf (hunk 1 above)
Comment (ank42) [reply l2/main.tf]: fixed

File: modules/vpc/output.tf (hunk 3 above)
Comment (ank42) [reply modules/vpc/output.tf]: fixed

File: modules/vpc/variable.tf (hunk 4 above)
Comment (ank42) [reply modules/vpc/variable.tf]: added new line!

File: modules/vpc/main.tf (hunk 2 above)
Comment (ank42) [reply modules/vpc/main.tf]: tried to solve it please check!

**Pull request merged**